tests/scripts/task_cpp_unittest.sh

```diff
@@ -45,20 +45,23 @@ python3 tests/scripts/task_build.py \
     --cmake-target cpptest \
     --build-dir "${BUILD_DIR}"
 
-# crttest requires USE_MICRO to be enabled, which is currently the case
-# with all CI configs
-pushd "${BUILD_DIR}"
-ninja crttest
-popd
+# crttest requires USE_MICRO to be enabled.
+if grep -Fq "set(USE_MICRO ON)" ${BUILD_DIR}/config.cmake; then
```

Comment:
i really wanted to be able to have you just look this up in CMakeCache.txt, but unfortunately we don't cache these. i wonder if we should be setting this with `set(USE_ABC ___ CACHED)` so we can inspect that. what if we modified [Summary.cmake](https://github.com/apache/tvm/blob/main/cmake/utils/Summary.cmake) to write the options to a file via https://cmake.org/cmake/help/latest/command/file.html#append?

Reply:
it might also be cleaner to split this out into `task_cpp_unittest_micro.sh` or something and just add it as another step everywhere that currently runs the cpp unit tests, that way its easy to run regardless of the specific build options

Reply:
I went the route of separating the script into two

Reply:
Update: It seems that there are `crt` tests that get included in the cpp test suite when `USE_MICRO` is enabled. 

Instead of separating those, I am following @areusch advice and writing to a file all the options used to build TVM.